saleor/checkout/complete_checkout.py

```diff
@@ -120,12 +121,17 @@ def _process_voucher_data_for_order(checkout_info: "CheckoutInfo") -> dict:
         return {}
 
     if voucher.usage_limit:
-        increase_voucher_usage(voucher)
+        # type-ignore added as we ensure that voucher code instance is created
```

Comment:
Instead of type ignore, which will hide you all arg-type problems. Can't you make:
```python
    code = cast(VoucherCode, code)
    if voucher.usage_limit:
        increase_voucher_usage(voucher, code)
    if voucher.apply_once_per_customer:
        customer_email = cast(str, checkout_info.get_customer_email())
        add_voucher_usage_by_customer(
            voucher, code, customer_email
        )
```